Source/CarthageKit/MachHeader.swift

```diff
@@ -70,14 +70,25 @@ extension MachHeader {
 		// This is the command `otool -h` actually invokes
 		let task = Task("/usr/bin/xcrun", arguments: [
 			"objdump",
-			"-macho",
-			"-private-header",
-			"-non-verbose",
+			"--macho",
+			"--private-header",
+			"--non-verbose",
 			url.resolvingSymlinksInPath().path,
 			]
 		)
 
 		return task.launch(standardInput: nil)
+            .flatMapError { _ -> SignalProducer<TaskEvent<Data>, TaskError> in
```

Comment:
@mikchmie So to recap my understanding: You are
- launching the `Task` with the new `--` prefix option
- in case that fails assume that it failed because it still expects the old `-` prefix
- so you "restart" the `Task` but now with the old implementation

I think we should not just expect that the error is due to the new prefix. How about instead we check the currently installed version of the objdump via `objdump --version` which outputs something along the line of this for Command Line Tools bundled with Xcode 13.3.1:
```
Apple LLVM version 13.1.6 (clang-1316.0.21.2.3)
```
If the version is greater than or equal to 13.1 (or whatever version already works with it) we use the double-dash prefix, if not we use the single-dash prefix. Wdyt?

Reply:
@Ro-M I've updated the implementation according to my proposal above. Please let me know what you think about it.

Reply:
I've digged a bit deeper. The documentation seems to be incomplete for older versions.

LLVM `3.8.0` is the oldest version working with double-dash (which translates to Xcode `7.3`). 
In other words LLVM `3.7.0` (Xcode `7.2.1`) is the newest NOT working version. 

However LLVM `3.7.0` doesn't have the `-private-header` option - only `-private-headers` (with an 's'), which has a totally different output. LLVM `3.8.0` does have that option. So to me it seems like we don't need any fallback mechanism after all, because our command wouldn't work in `3.7.0` either way.

There are even 2 comments about Xcode `7.x` incompatibility in the `MachHeader.swift` file:
Line 64:
>	 Note: `objdump` invocation requires flags unimplemented in Xcode 7.X ⋯ as it stands,
	         only a moreso-alternative codepath invokes this; be cautious to not invert that.
	         
Line 68:
> TODO: Potentially, `mdfind` other non-version-7.X Xcodes that might contain valid `objdump`.

BTW I've realized that in Apple's builds `objdump --version` prints "Clang version" instead of "LLVM version" which used to be quite different in the past (e.g. check Xcode 11 in the [table](https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_-_13.x_.28since_SwiftUI_framework.29_2) - LLVM `8.0.0` vs Clang `11.0.0`).

So to sum up - I think I can reset this PR to its initial commit.

Reply:
> However LLVM 3.7.0 doesn't have the -private-header option - only -private-headers (with an 's'), which has a totally different output. LLVM 3.8.0 does have that option. So to me it seems like we don't need any fallback mechanism after all, because our command wouldn't work in 3.7.0 either way.

There is `--private-headers` (alias `-p` which is an option you can use for `objdumb` on its own (similar to `--macho`) and there is `--private-header` (NO alias) which is ONLY available as a sub-option for the `--macho` command (`MACH-O ONLY OPTIONS AND COMMANDS`)

Reply:
But the current implementation doesn't use the `-private-headers` / `-p` option, so it doesn't matter, does it?

Running the current command (with single-dash syntax) using `objdump` `3.7.0`:
```
./objdump -macho -private-header -non-verbose ./GooglePlaces
```
returns an error:
```
objdump: Unknown command line argument '-private-header'.  Try: './objdump -help'
objdump: Did you mean '-private-headers'?
```

So it seems to me that changing single-dash to double-dash syntax will not introduce any regression (because this code doesn't currently  work with LLVM `< 3.8.0` either - the only difference will be a single character in the error message). Am I wrong?

Reply:
I was only pointing out the differences in the two private-header options as that seemed to cause confusion on your side (it did initially for me 😄)